lib/gems/pending/openstack/openstack_handle/handle.rb

```diff
@@ -197,6 +197,17 @@ def connect(options = {})
 
         raw_service = self.class.raw_connect_try_ssl(username, password, address, port, service, opts, api_version,
                                                      security_protocol)
+
+        # need to check if this is versionless keystone endpoint
+        if service == "Identity"
+          identity_prefix = (api_version == "v2" ? "v2.0" : "v3")
+          unless raw_service.credentials[:openstack_management_url].include?(identity_prefix)
+            opts[:openstack_identity_prefix] = identity_prefix
+            raw_service = self.class.raw_connect_try_ssl(username, password, address, port, service, opts, api_version,
```

Comment:
I wonder if (in a completely separate PR) the `raw_connect_try_ssl` could be refactored away entirely since I think we do a decent job of collecting whether the user is using SSL or not.  Whereas before we sort of had to guess...

Again, not for this PR.